tests/unit/conditions/test_return_value.py

```diff
@@ -23,6 +23,47 @@
 from nucypher.policy.conditions.lingo import ReturnValueTest
 
 
+def test_return_value_key():
+    test = ReturnValueTest(comparator=">", value="0", key="james")
+    assert test.eval({"james": 1})
+    assert not test.eval({"james": -1})
+
+    with pytest.raises(KeyError):
+        test.eval({"bond": 1})
+
+    test = ReturnValueTest(comparator=">", value="0", key=4)
+    assert test.eval({4: 1})
+    assert not test.eval({4: -1})
+
+    with pytest.raises(KeyError):
+        test.eval({5: 1})
+
+
+def test_return_value_index():
+    test = ReturnValueTest(comparator=">", value="0", key=0)
+    assert test.eval([1])
+    assert not test.eval([-1])
+
+    test = ReturnValueTest(comparator="==", value='"james"', key=3)
+    assert test.eval([0, 1, 2, '"james"'])
+
+    with pytest.raises(IndexError):
+        test.eval([0, 1, 2])
+
+
+def test_return_value_index_tuple():
+    test = ReturnValueTest(comparator=">", value="0", key=0)
+    assert test.eval((1,))
+    assert not test.eval((-1,))
+
+
+def test_return_value_with_context_variable_key_cant_run_eval():
```

Comment:
Nice - 👍 . Can we also add a test that includes actual context variable replacement for key?

See context variable replacemet tests in `tests/acceptance/blockchain/conditions/test_conditions.py`

Reply:
We should add tests for when a data type that is not list, tuple, or dict, to ensure the code handles it appropriately.

Reply:
> if key is supplied, and the value is not in list, tuple, or dict, then the value is just returned as-is

Feels like this case should fail. Key is optional so if supplied, it should presumably be adhered to...? Usage of key to get value from other data types that don't allow it (basically invalid use of key given the data type returned) eg. the value returned for comparison testing is an `int`, and can't use the `key` on it - this should presumably also be an exception